x11rb-protocol/src/errors.rs

```diff
@@ -76,6 +76,50 @@ impl fmt::Display for ParseError {
     }
 }
 
+/// An error that occurred while parsing the `$DISPLAY` environment variable
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[non_exhaustive]
+pub enum DisplayParsingError {
+    /// No explicit display was provided and the `$DISPLAY` environment variable is not set.
+    DisplayNotSet,
+
+    /// The given value could not be parsed. The input to parsing is provided.
+    MalformedValue(alloc::string::String),
```

Comment:
Can we do `Box<str>` instead of `String`? It would save a word of space.

Reply:
Done!